src/app/services/axles.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
+import { Axle, AxleSpacing } from '@models/vehicle-tech-record.model';
+import cloneDeep from 'lodash.clonedeep';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AxlesService {
+  normaliseAxles(axles?: Axle[], axleSpacings?: AxleSpacing[]): [Axle[] | undefined, AxleSpacing[] | undefined] {
+    let newAxles = cloneDeep(axles);
+    let newAxleSpacings = cloneDeep(axleSpacings);
+
+    if (axles && axleSpacings) {
+      if (axles.length > axleSpacings.length + 1) {
+        newAxleSpacings = this.generateAxleSpacing(axles.length, true, axleSpacings);
+      } else if (axles.length < axleSpacings.length + 1) {
+        newAxles = this.generateAxlesFromAxleSpacings(axleSpacings.length, axles);
+      }
+    } else if (axles && !axleSpacings) {
+      newAxleSpacings = this.generateAxleSpacing(axles.length);
+    } else if (!axles && axleSpacings?.length) {
+      newAxles = this.generateAxlesFromAxleSpacings(axleSpacings.length);
+    }
```

Comment:
```suggestion
  normaliseAxles(axles?: Axle[], axleSpacings?: AxleSpacing[]): [Axle[], AxleSpacing[]] {
    let newAxles = cloneDeep(axles ?? []);
    let newAxleSpacings = cloneDeep(axleSpacings ?? []);
    if (axles.length > axleSpacings.length + 1) {
      newAxleSpacings = this.generateAxleSpacing(axles.length, axleSpacings);
    } else if (axles.length < axleSpacings.length + 1) {
      newAxles = this.generateAxlesFromAxleSpacings(axleSpacings.length, axles);
    }
```
There doesn't seem to be scenario where you use the `setOriginalValues` and the `axleSpacingOriginal` independently in the `generateAxleSpacing` function so I'm not sure  why you would need both. 

I'm not saying this suggested change is 100%, just feels like there's something there

Reply:
Some great suggestion mate, I've applied this too, cheers 🙂